donut/modules/newsgroups/routes.py

```diff
@@ -0,0 +1,144 @@
+import flask
+from flask import jsonify
+from donut.modules.groups import helpers as groups
+from donut.modules.account import helpers as account
+from donut.modules.core.helpers import get_name_and_email
+from donut import auth_utils
+
+from . import blueprint, helpers
+
+
+@blueprint.route('/newsgroups')
+def newsgroups_home():
+    if 'username' not in flask.session:
+        return flask.abort(403)
+    return flask.render_template(
+        'newsgroups.html', groups=helpers.get_newsgroups())
+
+
+@blueprint.route('/newsgroups/post')
+@blueprint.route('/newsgroups/post/<group_id>')
+def post(group_id=None):
+    if 'username' not in flask.session:
+        return flask.abort(403)
+    user_id = auth_utils.get_user_id(flask.session['username'])
+    if not group_id:
+        group_id = flask.request.form.get('group')
+    return flask.render_template(
+        'post.html',
+        groups=helpers.get_my_newsgroups(user_id, True),
+        group_selected=group_id)
+
+
+@blueprint.route('/newsgroups/postmessage', methods=['POST'])
+def post_message():
+    if 'username' not in flask.session:
+        return flask.abort(403)
+    user_id = auth_utils.get_user_id(flask.session['username'])
+    fields = ['group', 'subject', 'msg', 'poster']
+    data = {}
+    for field in fields:
+        data[field] = flask.request.form.get(field)
+    data['group_name'] = groups.get_group_data(data['group'],
+                                               ['group_name'])['group_name']
+    if not data['poster']:
+        data['poster'] = get_name_and_email(user_id)['full_name']
+    else:
+        data['poster'] = ' '.join((data['group_name'], data['poster']))
+    if helpers.send_email(data):
+        flask.flash('Email sent')
+        helpers.insert_email(user_id, data)
+    else:
+        flask.flash('Email failed to send')
+    return flask.redirect(flask.url_for('newsgroups.post'))
+
+
+@blueprint.route('/newsgroups/viewgroup/<group_id>')
+def view_group(group_id):
+    if 'username' not in flask.session:
+        return flask.abort(403)
+    user_id = auth_utils.get_user_id(flask.session['username'])
+    pos_actions = helpers.get_user_actions(user_id, group_id)
+    fields = [
+        'group_id', 'group_name', 'group_desc', 'anyone_can_send',
+        'members_can_send', 'visible', 'admin_control_members'
+    ]
+    group_info = groups.get_group_data(group_id, fields)
+    applications = None
+    if pos_actions and pos_actions['control']:
+        applications = helpers.get_applications(group_id)
+
+    return flask.render_template(
+        'group.html',
+        group=group_info,
+        member=groups.is_user_in_group(user_id, group_id),
+        actions=pos_actions,
+        messages=helpers.get_past_messages(group_id),
```

Comment:
Should be restricted to users in group

Reply:
Aren't groups visible to everyone as long as visible is true?

Reply:
But the messages should be hidden